src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -29,8 +29,37 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
     MTRDeviceStateUnreachable = 2
 };
 
+/**
+ * This bitmask is used to specify the type of logs requested from this device.
+ *
+ * The logs types are : End User Support, Network Diagnostics and Crash logs.
+ */
+typedef NS_OPTIONS(NSUInteger, MTRDiagnosticLogType) {
+    MTRDiagnosticLogTypeEndUserSupport = 1 << 0, // End user support logs are requested
+    MTRDiagnosticLogTypeNetworkDiagnostics = 1 << 1, // Network Diagnostics logs are requested
+    MTRDiagnosticLogTypeCrash = 1 << 2, // Crash logs are requested
+    MTRDiagnosticLogTypesAll = // All the above log types are requested
+    MTRDiagnosticLogTypeEndUserSupport | MTRDiagnosticLogTypeNetworkDiagnostics | MTRDiagnosticLogTypeCrash,
+    MTRDiagnosticLogTypeUnknown = 1 << 3,
+} MTR_PROVISIONALLY_AVAILABLE;
+
 @protocol MTRDeviceDelegate;
 
+MTR_PROVISIONALLY_AVAILABLE
+@interface MTRDiagnosticLogResult : NSObject
+
+/**
+ * A boolean indicating that the request for diagnostic logs has timed out.
+ */
+@property (readonly, nonatomic) BOOL requestTimedOut;
+
+/**
+ * Array containing the file path(s) for the desired log type(s)
+ */
+@property (readonly, nonatomic, nullable) NSArray<NSString *> * filePathArray;
```

Comment:
Should this really be nullable?  I'd think we would just have an empty array if we have nothing....

Reply:
done.